src/main/kotlin/org/malachite/estella/aplication/api/ApplicationController.kt

```diff
@@ -69,46 +67,41 @@ class ApplicationController(
             .let { ResponseEntity.ok(it) }
 
     @CrossOrigin
-    @GetMapping("/job-seeker/{jobSeekerId}")
-    fun getAllApplicationsByJobSeeker(@PathVariable jobSeekerId: Int) =
-            applicationService.getApplicationsByJobSeeker(jobSeekerId)
+    @GetMapping("/job-seeker")
+    fun getAllApplicationsByJobSeeker(@RequestHeader(EStellaHeaders.jwtToken) jwt: String?)
+            : ResponseEntity<Any> =
+        securityService.getJobSeekerFromJWT(jwt)
+            ?.id
+            ?.let { applicationService.getApplicationsByJobSeeker(it) }
+            ?.let { ResponseEntity.ok(it) }
+            ?: UNAUTH
 
-    @CrossOrigin
-    @DeleteMapping("/delete/{applicationId}")
-    fun deleteApplication(@PathVariable applicationId: Int) =
-        applicationService.deleteApplication(applicationId)
 
     @CrossOrigin
     @PutMapping("/{applicationId}/next")
-    fun updateApplicationStage(@RequestHeader(EStellaHeaders.jwtToken) jwt: String?, @PathVariable applicationId: Int): ResponseEntity<Message> {
+    fun updateApplicationStage(
+        @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
+        @PathVariable applicationId: Int
+    ): ResponseEntity<Any> =
         applicationService.getApplicationById(applicationId).let {
-            recruitmentProcessService.getProcessFromStage(it.stage).let {
-                if (!checkUserRights(it.offer, jwt)) return OwnResponses.UNAUTH
-                return applicationService.setNextStageOfApplication(applicationId).let { OwnResponses.SUCCESS }
-            }
+            recruitmentProcessService.getProcessFromStage(it.stage)
+        }.let {
+            if (!securityService.checkOfferRights(it.offer, jwt)) return UNAUTH
+            applicationService.setNextStageOfApplication(applicationId).let { SUCCESS }
         }
-    }
+
 
     @CrossOrigin
     @PutMapping("/{applicationId}/reject")
-    fun rejectApplication(@RequestHeader(EStellaHeaders.jwtToken) jwt: String?, @PathVariable applicationId: Int): ResponseEntity<Message> {
-        applicationService.getApplicationById(applicationId).let {
-            recruitmentProcessService.getProcessFromStage(it.stage).let {
-                if (!checkUserRights(it.offer, jwt)) return OwnResponses.UNAUTH
-                return applicationService.rejectApplication(applicationId).let { OwnResponses.SUCCESS }
-
-            }
+    fun rejectApplication(
+        @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
+        @PathVariable applicationId: Int
+    ): ResponseEntity<Any> =
+        applicationService.getApplicationById(applicationId)
+            .let {
+                recruitmentProcessService.getProcessFromStage(it.stage) }
+            .let {
+                if (!securityService.checkOfferRights(it.offer, jwt)) return UNAUTH
```

Comment:
securityService? Nie mieliśmy robić takich sprawdzeń w odpowiednich serwisach czyli tutaj applicationService?

Reply:
Ogólnie to jest dopiero robione przeze mnie, więc albo to się poprawi u mnie, albo jak szybciej zmerguję, to wtedy to Tomek poprawi.

Reply:
generalnie chciałem to dostoswować po zmergowaniu refactoru Dawida, żeby się wykluła jakaś konwencja

Reply:
Ogólnie nie wiem, czy jest sens na to czekać, bo teraz jeszcze Kuba przy tym grzebie i nie do końca wiadomo, co się z tego wykluje.

Reply:
w takim narazie narazie zostawie tak jak jest, żeby potem nie zmieniać po kilka razy jak się okaże co z tym springiem